pkg/awsutils/awsutils.go

```diff
@@ -685,14 +702,33 @@ func awsUtilsErrInc(fn string, err error) {
 	awsUtilsErr.With(prometheus.Labels{"fn": fn, "error": err.Error()}).Inc()
 }
 
+func exponentialBackoff(retryFunc func() error) error {
+	settings := backoff.NewExponentialBackOff()
+	settings.MaxElapsedTime = exponentialBackoffMaxElapsedTime
+
+	return backoff.Retry(retryFunc, settings)
+}
+
 // FreeENI detachs ENI interface and delete ENI interface
 func (cache *EC2InstanceMetadataCache) FreeENI(eniName string) error {
 	log.Infof("Trying to free eni: %s", eniName)
 
 	// find out attachment
 	//TODO: use metadata
-	_, attachID, err := cache.DescribeENI(eniName)
-	if err != nil {
+	var attachID *string
+	var err error
+	describeEniFunc := func() error {
+		_, attachID, err = cache.DescribeENI(eniName)
+		if err != nil {
+			// If the ENI doesn't exist, do not attempt to backoff, just return permanent error
+			if containsAttachmentIDNotFoundError(err) {
```

Comment:
I don't think it will ever return this error for an ENI, as its not looking for an attachment id? You prob want some form of a ResourceNotFound error